implementations/elixir/ockam/ockam/lib/ockam/transport/tcp/address.ex

```diff
@@ -1,87 +1,60 @@
 defmodule Ockam.Transport.TCPAddress do
-  @moduledoc false
+  @moduledoc """
+  Functions to work with TCP transport address
+  """
+  alias Ockam.Address
 
-  alias __MODULE__
+  @address_type 1
 
-  defstruct [:ip, :port]
+  @type t :: Address.t(1)
 
-  # tcp address type
-  @tcp 1
+  def type(), do: @address_type
 
-  # ip address type tags
-  @ipv4 0
-  @ipv6 1
+  @spec to_host_port(t()) :: {:ok, {String.t(), integer()}} | {:error, any()}
+  def to_host_port(address) do
+    case is_tcp_address(address) do
+      true ->
+        parse_host_port(Address.value(address))
 
-  @spec deserialize(any) ::
-          {:error, {:not_a_valid_serialized_tcp_address, any}}
-          | %Ockam.Transport.TCPAddress{
-              ip: {byte, byte, byte, byte} | {byte, byte, byte, byte, byte, byte, byte, byte},
-              port: char
-            }
-
-  def deserialize(%{type: @tcp, value: value}), do: deserialize(value)
-
-  def deserialize(value) when is_list(value), do: deserialize(IO.iodata_to_binary(value))
-
-  def deserialize(<<@ipv4::8, a::8, b::8, c::8, d::8, port::unsigned-little-integer-16>>) do
-    %TCPAddress{ip: {a, b, c, d}, port: port}
-  end
-
-  def deserialize(
-        <<@ipv6::8, a::8, b::8, c::8, d::8, e::8, f::8, g::8, h::8,
-          port::unsigned-little-integer-16>>
-      ) do
-    %TCPAddress{ip: {a, b, c, d, e, f, g, h}, port: port}
+      false ->
+        {:error, {:invalid_address_type, Address.type(address)}}
+    end
   end
 
-  def deserialize(value), do: {:error, {:not_a_valid_serialized_tcp_address, value}}
-end
-
-defimpl Ockam.Address, for: Ockam.Transport.TCPAddress do
-  alias Ockam.Transport.TCPAddress
+  def parse_host_port(value) do
+    case Regex.split(~r/(:)\d*$/, value, include_captures: true, on: :all_but_first, trim: true) do
+      [host, ":", port_str] ->
+        case Integer.parse(port_str) do
+          {port, ""} ->
+            {:ok, {host, port}}
 
-  def type(_address), do: 1
-  def value(address), do: address
-end
-
-defimpl Ockam.Serializable, for: Ockam.Transport.TCPAddress do
-  alias Ockam.Transport.TCPAddress
-
-  # tcp address type
-  @tcp 1
-
-  # ip address type tags
-  @ipv4 0
-  @ipv6 1
+          _other ->
+            {:error, {:invalid_port, port_str}}
+        end
 
-  def serialize(%TCPAddress{ip: ip, port: port}) do
-    with {:ok, serialized_ip} <- serialize_ip(ip),
-         {:ok, serialized_port} <- serialize_port(port) do
-      %{type: @tcp, value: :binary.list_to_bin([serialized_ip, serialized_port])}
+      _other ->
```

Comment:
Do we always expect an explicit port or, could it happen that the user assumes a default port?